webhook/main.go

```diff
@@ -0,0 +1,137 @@
+	ctx := context.Background()
+
+	webhookSecret, err := os.ReadFile("/etc/secrets/webhook/key")
+	if err != nil {
+		log.Printf("failed to read webhook secret: %v", err)
+		resp.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	payload, err := github.ValidatePayload(req, webhookSecret)
+	if err != nil {
+		log.Printf("failed to validate payload: %v", err)
+		resp.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	event, err := github.ParseWebHook(github.WebHookType(req), payload)
+	if err != nil {
+		log.Printf("failed to parse webhook: %v", err)
+		resp.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	switch event := event.(type) {
+	case *github.WorkflowJobEvent:
+		if event.Action == nil || *event.Action != "queued" {
+			log.Printf("no action taken for %v", *event.Action)
+			resp.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		kmsClient, err := kms.NewKeyManagementClient(ctx)
+		if err != nil {
+			log.Printf("failed to create kms client: %v", err)
+			resp.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		keyId := os.Getenv("KEY_ID")
+		signer, err := gcpkms.NewSigner(ctx, kmsClient, keyId)
+		if err != nil {
+			log.Printf("failed to create app signer: %v", err)
+			resp.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		appId := os.Getenv("APP_ID")
+		app, err := githubauth.NewApp(appId, signer)
+		if err != nil {
+			log.Printf("failed to setup client: %v", err)
+			resp.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		installation, err := app.InstallationForID(ctx, strconv.FormatInt(*event.Installation.ID, 10))
+		if err != nil {
+			log.Printf("failed to setup client: %v", err)
+			resp.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		httpClient := oauth2.NewClient(ctx, installation.AllReposOAuth2TokenSource(ctx, map[string]string{
+			"administration": "write",
+		}))
+		gh := github.NewClient(httpClient)
+
+		// Note that even though event.WorkflowJob.RunID is used for a dynamic string, it's not
+		// guaranteed that particular job will run on this specific runner.
+		jitconfig, _, err := gh.Actions.GenerateRepoJITConfig(ctx, *event.Org.Login, *event.Repo.Name, &github.GenerateJITConfigRequest{Name: fmt.Sprintf("GCP-%d", event.WorkflowJob.RunID), RunnerGroupID: 1, Labels: []string{"self-hosted", "Linux", "X64"}})
+		if err != nil {
+			log.Printf("failed to generate jitconfig: %v", err)
+			resp.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		client, err := cloudbuild.NewClient(ctx)
+		if err != nil {
+			log.Printf("failed to create cloudbuild client: %v", err)
+			resp.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		projectId := os.Getenv("PROJECT_ID")
+		triggerId := os.Getenv("TRIGGER_ID")
+		buildReq := &cloudbuildpb.RunBuildTriggerRequest{
+			Name:      fmt.Sprintf("projects/%s/locations/us-west1/triggers/github-actions-runner", projectId),
```

Comment:
The location and maybe the trigger name should be env vars or command line options. This shouldn't be hard coded.